### PR TITLE
API for fetching nearby coordinates information

### LINKED ISSUE #1934: ability to refresh displayed inline maps' displayed data when you pan/zoom
### Please describe the problem (or idea)

You can embed inline maps using this inline powertag: https://publiclab.org/wiki/inline-maps

The code driving this is here:

https://github.com/jywarren/plots2/blob/184eced60d9d6842cad8df469bb1df2838528522/app/models/concerns/node_shared.rb#L102-L124

And the template is here: https://github.com/jywarren/plots2/blob/master/app/views/map/_leaflet.html.erb

However, when you drag the map, the markers aren't refreshed for the new viewing area. 

Let's break this in three:

1. [x] let's make an API call for data near a point, similar to https://publiclab.org/api/srch/notes?srchString=foo but perhaps https://publiclab.org/api/srch/notes?geo=41.0,-91.3. This could be built out from https://github.com/publiclab/plots2/blob/master/app/api/srch/search.rb#L52-L72, or a new method could be made
2. [x] let's rewire https://github.com/jywarren/plots2/blob/184eced60d9d6842cad8df469bb1df2838528522/app/models/concerns/node_shared.rb#L102-L124 to load a blank map which then loads the API call for the center point, and displays notes as points on hearing back from the API
3. [x] let's set the inline map to re-fetch notes near its center point each time the map is panned or zoomed

This is a big multi-part issue that we can solve in three or more steps. Anyone interested in breaking out the first one?

### PR DESCRIPTION
**Pull request by Gauravano**
Fixes #1934 
API working.Suggestions regarding naming and comments are welcome.

### CODE REVIEW

**Comment by Gauravano** [comment]
When correct format is followed in URL
first

When correct format not followed
second

File: app/services/search_service.rb [hunk 1]
```diff
@@ -182,4 +182,29 @@ def textSearch_questions(srchString)
     end
     sresult
   end
+
+  def search_nearbyLocations(srchString)
```
Comment (jywarren) [inline app/services/search_service.rb]: This is looking great! Could we call this `nearbyNodes`?

File: app/api/srch/search.rb [hunk 2]
```diff
@@ -115,6 +115,30 @@ class Search < Grape::API
         sresult
       end
 
+      # Request URL should be /api/srch/locations?srchString=QRY[&seq=KEYCOUNT&showCount=NUM_ROWS&pageNum=PAGE_NUM]
```
Comment (jywarren) [inline app/api/srch/search.rb]: Super! Perhaps we can add this also to `/doc/API.md`?

File: app/services/search_service.rb (hunk 1 above)
Comment (Gauravano) [reply app/services/search_service.rb]: doing changes now

File: app/api/srch/search.rb (hunk 2 above)
Comment (Gauravano) [reply app/api/srch/search.rb]: Done

File: app/services/search_service.rb [hunk 3]
```diff
@@ -182,4 +182,29 @@ def textSearch_questions(srchString)
     end
     sresult
   end
+
+  def nearbyNodes(srchString)
+    sresult = DocList.new
+    coordinates = srchString.split(",")
+    lat = coordinates[0]
+    lon = coordinates[1]
+
```
Comment (asahu8) [inline app/services/search_service.rb]: latitude, longitude = coordinates 
directly we can assign, ruby will take care of it.

File: app/services/search_service.rb [hunk 4]
```diff
@@ -182,4 +182,29 @@ def textSearch_questions(srchString)
...
+    nids = NodeTag.joins(:tag)
+               .where('name LIKE ?', 'lat:' + lat[0..lat.length - 2] + '%')
+               .collect(&:nid)
+
+    nids = nids || []
+
```
Comment (asahu8) [inline app/services/search_service.rb]: nids ||= []

File: app/api/srch/search.rb [hunk 5]
```diff
@@ -115,6 +115,30 @@ class Search < Grape::API
...
+      params do
+        requires :srchString, type: String, documentation: { example: 'Spec' }
+        optional :seq, type: Integer, documentation: { example: 995 }
+        optional :showCount, type: Integer, documentation: { example: 3 }
+        optional :pageNum, type: Integer, documentation: { example: 0 }
+      end
+      get :locations do
+        sresult = DocList.new
+        unless params[:srchString].nil? || params[:srchString] == 0 || !(params[:srchString].include? ",")
+          sservice = SearchService.new
```
Comment (asahu8) [inline app/api/srch/search.rb]: params[:srchString].nil? || params[:srchString] == 0
this particular code is kind of used in multiple places in this file. can't be move it to a private method and re-use it relevant place ?

**Comment by Gauravano** [comment]
jeff could you plz help with the build logs 
build

File: app/services/search_service.rb (hunk 3 above)
Comment (Gauravano) [reply app/services/search_service.rb]: You are absolutely right but I think for sake of clarity of all writing one extra line is good

**Comment by jywarren** [comment]
hmm, this is odd, no? let me take a look.

**Comment by jywarren** [comment]
`Cannot find a merge base between danger_base and danger_head` -- well, searching for this while restarting the build.

**Comment by jywarren** [comment]
https://github.com/danger/danger/issues/768

**Comment by jywarren** [comment]
oh, do you have two branches of almost the same name?

File: app/api/srch/search.rb (hunk 5 above)
Comment (jywarren) [reply app/api/srch/search.rb]: I think so -- but perhaps independently from this PR so we address code repetition as its own issues. Would you mind opening a new issue to address this? I agree that the services and api code in particular can be enormously compacted without loss of organization.  Thank you!

File: app/services/search_service.rb (hunk 3 above)
Comment (jywarren) [reply app/services/search_service.rb]: Good suggestion! I always forget to do this myself.

File: app/services/search_service.rb (hunk 4 above)
Comment (jywarren) [reply app/services/search_service.rb]: :-)

**Comment by jywarren** [comment]
Tough one -- it says the name collision may originate in a different PR -- scary! I'll watch if other PR's travis tests are failing...

**Comment by Gauravano** [comment]
same logs again

**Comment by jywarren** [comment]
:-/ i'm not sure about this one! @publiclab/reviewers can anyone help debug this?

**Comment by Gauravano** [comment]
There is warning about missing secret_key_base too.Do you think it has any role to play?

**Comment by jywarren** [comment]
Hmm, that may be a question for @icarito ... It's a rails 4.1 thing!

**Comment by Gauravano** [comment]
Jeff error was generated due to the wrong merging of code on my master branch.Its resolved now.Thanks

**Pull request closed without merging**